app/src/main/java/org/thoughtcrime/securesms/conversation/start/NewConversationHomeFragment.kt

```diff
@@ -35,43 +42,76 @@ class NewConversationHomeFragment : Fragment() {
     @Inject
     lateinit var textSecurePreferences: TextSecurePreferences
 
-    lateinit var delegate: NewConversationDelegate
+    var delegate = MutableStateFlow<NewConversationDelegate?>(null)
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View = createThemedComposeView { NewConversationScreen() }
+    ): View = createThemedComposeView {
+        // Warning, brittle code, NewConversationScreen will not be visible if no delegate is set
+        // before onCreateView is called.
+        delegate.collectAsState().value?.let {
```

Comment:
I don't understand the logic here.
What is this delegate and why is there a scenario possible in which we wouldn't have a screen?

Reply:
it's just callbacks. simplified. hardened. fixed.